File: src/Entity/Chambre.php
<?php

namespace App\Entity;

use App\Repository\ChambreRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use JsonSerializable;
use Symfony\Component\Serializer\Annotation\Groups;
use App\Entity\ReservationMaison;

/**
 * @ORM\Entity(repositoryClass=ChambreRepository::class)
 */
class Chambre implements JsonSerializable
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     * @Groups("chambre")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="nom est requis")
     * @Groups("chambre")
     */
    private $nom;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="type de lit est requis")
     * @Groups("chambre")
     */
    private $typeLit;

    /**
     * @ORM\Column(type="integer")
     * @Assert\NotBlank(message="Nombre de personne est requis")
     * @Groups("chambre")
     */
    private $nbr_pers;

    /**
     * @ORM\Column(type="float")
     * @Assert\NotBlank(message="Prix est requis")
     * @Groups("chambre")
     */
    private $prix;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Photo est requis")
     * @Groups("chambre")
     */
    private $photo;

    /**
     * @ORM\ManyToOne(targetEntity=Maison::class, inversedBy="chambres")
     * @Groups("chambre")
     */
    private $maison;

    /**
     * @ORM\OneToMany(targetEntity=ReservationMaison::class, mappedBy="chambre")
     * @Groups("chambre")
     */
    private $reservations;

    /**
     * @ORM\Column(type="string", nullable=true)
     * @Groups("chambre")
     */
    private $user_id;



    public function __construct()
    {
        $this->reservations = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getTypeLit(): ?string
    {
        return $this->typeLit;
    }

    public function setTypeLit(string $typeLit): self
    {
        $this->typeLit = $typeLit;

        return $this;
    }

    public function getNbrPers(): ?int
    {
        return $this->nbr_pers;
    }

    public function setNbrPers(int $nbr_pers): self
    {
        $this->nbr_pers = $nbr_pers;

        return $this;
    }

    public function getPrix(): ?float
    {
        return $this->prix;
    }

    public function setPrix(float $prix): self
    {
        $this->prix = $prix;

        return $this;
    }

    public function getPhoto()
    {
        return $this->photo;
    }

    public function setPhoto( $photo)
    {
        $this->photo = $photo;

        return $this;
    }

    public function getMaison(): ?Maison
    {
        return $this->maison;
    }

    public function setMaison(?Maison $maison): self
    {
        $this->maison = $maison;

        return $this;
    }


    /**
     * @return Collection|ReservationMaison[]
     */
    public function getReservations(): Collection
    {
        return $this->reservations;
    }

    public function addReservations(ReservationMaison $reservations): self
    {
        if (!$this->reservations->contains($reservations)) {
            $this->reservations[] = $reservations;
            $reservations->setChambre($this);
        }

        return $this;
    }

    public function removeReservations(ReservationMaison $reservations): self
    {
        if ($this->reservations->removeElement($reservations)) {
            // set the owning side to null (unless already changed)
            if ($reservations->getChambre() === $this) {
                $reservations->setChambre(null);
            }
        }

        return $this;
    }

    public function jsonSerialize()
    {
        return array(
            'id' => $this->id,
            'nom' => $this->nom,
            'typeLit' => $this->typeLit,
            'nbr_pers' => $this->nbr_pers,
            'photo' => $this->photo,
            'prix' => $this->prix,
            'maison_id' => $this->maison,
            'nomHebergeur' => $this->user_id
        );
    }

    public function getUserId(): ?string
    {
        return $this->user_id;
    }

    public function setUserId(?string $user_id): self
    {
        $this->user_id = $user_id;

        return $this;
    }
}

File: models/MaisonDHote/Chambre.java
package Tfarhida.entities;

import javafx.beans.property.BooleanProperty;
import javafx.beans.property.SimpleBooleanProperty;
import javafx.scene.control.Button;
import javafx.scene.image.ImageView;

public class Chambre {
    
    private int id, nbr_pers, idM;
    private double prix;
    String nom, type_lit, photo;
    private final BooleanProperty reserver = new SimpleBooleanProperty();
    ImageView img;
    
    public Chambre(){
        
    }

    public Chambre(int id, int idM, String nom, String type_lit, int nbr_pers, double prix, String photo) {
        this.id = id;
        this.idM = idM;
        this.nom = nom;
        this.type_lit = type_lit;
        this.nbr_pers = nbr_pers;
        this.prix = prix;
        this.photo = photo;
       reserver.setValue(false);
 
    }
    
    public Chambre(String nom, double prix, String type_lit ){
        this.nom = nom;
        this.prix = prix;
        this.type_lit = type_lit;
    }

    public ImageView getImg() {
        return img;
    }

    public void setImg(ImageView img) {
        this.img = img;
    }

    public int getIdM() {
        return idM;
    }

    public void setIdM(int idM) {
        this.idM = idM;
    }
    
    

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getNbr_pers() {
        return nbr_pers;
    }

    public void setNbr_pers(int nbr_pers) {
        this.nbr_pers = nbr_pers;
    }

    public double getPrix() {
        return prix;
    }

    public void setPrix(double prix) {
        this.prix = prix;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getType_lit() {
        return type_lit;
    }

    public void setType_lit(String type_lit) {
        this.type_lit = type_lit;
    }

    public String getPhoto() {
        return photo;
    }

    public void setPhoto(String photo) {
        this.photo = photo;
    }

    public boolean isReserver() {
        return reserver.getValue();
    }

    public void setReserver(boolean reserver) {
        this.reserver.setValue(reserver);
    }

  
    

    @Override
    public String toString() {
        return "Chambre{" + "id=" + id + ", nbr_pers=" + nbr_pers + ", prix=" + prix + ", nom=" + nom + ", type_lit=" + type_lit + ", photo=" + photo + '}';
    }
    
    
    
    
}
